ocean/transects/compute_transects.py

```diff
@@ -31,41 +32,111 @@ def get_mask_short_names(mask):
+  nEdgesInTransect = np.zeros(nTransects)
+  edgeVals = np.zeros((nTransects,maxEdges))
+  for i in range(nTransects):
+    amask = mask.sel(shortNames=transectList[i]).squeeze()
+    transectEdges = amask.transectEdgeGlobalIDs.values
+    inds = np.where(transectEdges > 0)[0]
+    nEdgesInTransect[i] = len(inds)
+    transectEdges = transectEdges[inds]
+    edgeVals[i,:len(inds)] = np.asarray(transectEdges-1, dtype='i')
 
-  plt.figure()
-  plt.plot(t, transport, 'k-', lw=3, label='monthly-averaged transport')
-  plt.gca().fill_between(t, (134.-14.)*np.ones_like(t), (134.+14.)*np.ones_like(t), alpha=0.3, label='observations')
-  plt.legend(loc='best', frameon=False)
-  plt.ylabel('Transport (Sv)')
-  plt.xlabel('yrs')
-  plt.title('Transport for ' + name)
-  plt.savefig('transport' + name.replace(' ', '') +'.png')
+  nEdgesInTransect = np.asarray(nEdgesInTransect, dtype='i')
+# Create a list with the start and stop for transect bounds
+  nTransectStartStop = np.zeros(nTransects+1)
+  for j in range(1,nTransects+1):
+    nTransectStartStop[j] = nTransectStartStop[j-1] + nEdgesInTransect[j-1]
 
+  edgesToRead = edgeVals[0,:nEdgesInTransect[0]]
+  for i in range(1,nTransects):
+    edgesToRead = np.hstack([edgesToRead,edgeVals[i,:nEdgesInTransect[i]]])
+
+  edgesToRead = np.asarray(edgesToRead, dtype='i')
+
+  dvEdge = mesh.dvEdge.sel(nEdges=edgesToRead).values
+
+  edgeSigns = np.zeros((nTransects,len(edgesToRead)))
+  for i in range(nTransects):
+    edgeSigns[i,:] = mask.sel(nEdges=edgesToRead, shortNames=transectList[i]).squeeze().transectEdgeMaskSigns.values
+
+# Read time average files one at a time and slice
+  fileList = sorted(glob.glob(timeavg))
+  transport = np.zeros((len(fileList),nTransects))
+  t = np.zeros(len(fileList))
+  for i,fname in enumerate(fileList):
+    ncid = Dataset(fname,'r')
+    vel = ncid.variables['timeMonthly_avg_normalTransportVelocity'][0,edgesToRead,:]
+    t[i] = ncid.variables['timeMonthly_avg_daysSinceStartOfSim'][:] / 365.
+    ncid.close()
+#   Compute transport for each transect
+    for j in range(nTransects):
+      start = int(nTransectStartStop[j])
+      stop = int(nTransectStartStop[j+1])
+      transport[i,j] = (dvEdge[start:stop,np.newaxis]*h[np.newaxis,:]*vel[start:stop,:] \
+          *edgeSigns[j,start:stop,np.newaxis]).sum()*m3ps_to_Sv
+
+    print fname
+ 
+  for j in range(nTransects):
+    plt.figure()
+    plt.plot(t, transport[:,j], 'k-', lw=3, label='monthly-averaged transport')
+    if transectList[j] == 'Drake Passage':
+      plt.gca().fill_between(t, (134.-14.)*np.ones_like(t), (134.+14.)*np.ones_like(t), alpha=0.3, label='observations')
+    plt.legend(loc='best', frameon=False)
+    plt.ylabel('Transport (Sv)')
+    plt.xlabel('yrs')
+    plt.title('Transport for ' + name)
+    plt.savefig('transport' + transectList[j].replace(' ', '') +'.png')
+
+# Add calls to save transport and then can build up
+  ncid=Dataset('transport.nc',mode='w',clobber=True, format='NETCDF3_CLASSIC')
```

Comment:
This would be the location where the output file name could be inserted, for instance.

Reply:
good idea.  I made the change